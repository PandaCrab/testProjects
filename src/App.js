import React from 'react';
import { Outlet, Link } from 'react-router-dom';
import { useSelector } from 'react-redux';

import { OrderPlate } from './components/Index';

import {
  Header,
  HeaderLogo,
  HeaderText,
  ShoppingBasket,
  BasketContainer,
  Slash,
  Order,
  Nav,
  Basket,
  ShoppingBasketText,
  NumberOfStuff,
  CircleOfNumber
} from './styles/AppStyles';
import { GlobalStyles } from './GlobalStyles';

const App = () => {
  const stuff = useSelector(state => state.order.stuff);


  return (
    <>
      <GlobalStyles />
      <Header>
        <HeaderText> <HeaderLogo>&lt; <Slash>&frasl;</Slash>&gt;</HeaderLogo> Front-end Developer Test Task</HeaderText>
        <ShoppingBasket>
          <ShoppingBasketText>cart</ShoppingBasketText>
          <BasketContainer>
            <Basket /> 
            { stuff.length === undefined ?
              null
              :
              <CircleOfNumber><NumberOfStuff>{stuff.length}</NumberOfStuff></CircleOfNumber>
            }   
          </BasketContainer>
        </ShoppingBasket>
        
      </Header>
      <Order>
        <Nav>
          <Link to="/shipping">Shipping</Link> &gt;
          <Link to="/billing">Billing</Link> &gt;
          <Link to="/payment">Payment</Link>
        </Nav>
        <Outlet />
        <OrderPlate />
      </Order>
    </>
  );
}

export default App;
