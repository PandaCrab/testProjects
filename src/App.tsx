import React, { useEffect, useState } from 'react';
import { Outlet, useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { takeGeolocation, takeNavigagtorAddress } from './redux/ducks/address';
import { 
  Box, 
  Divider, 
  List, 
  ListItem, 
  ListItemButton, 
  ListItemIcon, 
  ListItemText, 
  SwipeableDrawer 
} from '@mui/material';

import { RootState, AppDispatch } from './types';

import {
  Header,
  HeaderLogo,
  HeaderText,
  ShoppingBasket,
  BasketContainer,
  Slash,
  Basket,
  ShoppingBasketText,
  NumberOfStuff,
  CircleOfNumber
} from './styles/AppStyles';
import { GlobalStyles } from './GlobalStyles';

const App = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch<AppDispatch>();
  const stuff = useSelector((state: RootState)=> state.order.stuff);

  const [viewDrawer ,setViewDrawer] = useState(false)

  useEffect(() => {
    if ('geolocation' in navigator) {
      navigator
        .geolocation
        .getCurrentPosition (
          position => {
            dispatch (
              takeGeolocation (
                position.coords.latitude, 
                position.coords.longitude
              )
            );
            dispatch (takeNavigagtorAddress());
          });
      } else {
        return;
      }
    });

  return (
    <>
      <GlobalStyles />
      <Header>
        <HeaderText> <HeaderLogo onClick={() => navigate('/')}>&lt; <Slash>&frasl;</Slash>&gt;</HeaderLogo> Front-end Developer Test Task</HeaderText>
        <ShoppingBasket>
          <ShoppingBasketText>cart</ShoppingBasketText>
          <BasketContainer>
            <Basket /> 
            { stuff && stuff.length > 0 && (
              <CircleOfNumber><NumberOfStuff>{stuff.length}</NumberOfStuff></CircleOfNumber>
            )}   
          </BasketContainer>
        </ShoppingBasket> 
      </Header>
      <SwipeableDrawer
        anchor={"left"}
        open={viewDrawer}
        onClose={() => setViewDrawer(false)}
        onOpen={() => setViewDrawer(true)}
      >
        <Box
          sx={{ width: 250 }}
          role="presentation"
          onClick={() => setViewDrawer(false)}
        >
          <List>
              <ListItem key='storage' disablePadding>
                <ListItemButton onClick={() => navigate('/storage')}>
                  <ListItemIcon>
                    { '@' }
                  </ListItemIcon>
                  <ListItemText primary='Products Storage' />
                </ListItemButton>
              </ListItem>
              <Divider />
              <ListItem key='shipping' disablePadding>
                <ListItemButton onClick={() => navigate('/shipping')}>
                  <ListItemIcon>
                    { '@' }
                  </ListItemIcon>
                  <ListItemText primary='Delivery info' />
                </ListItemButton>
              </ListItem>
              <Divider />
              <ListItem key='shopping' disablePadding>
                <ListItemButton onClick={() => navigate('/shopping')}>
                  <ListItemIcon>
                    { '@' }
                  </ListItemIcon>
                  <ListItemText primary='Start shopping' />
                </ListItemButton>
              </ListItem>
              <Divider />
              <ListItem key='chart' disablePadding>
                <ListItemButton onClick={() => navigate('/chart')}>
                  <ListItemIcon>
                    { '@' }
                  </ListItemIcon>
                  <ListItemText primary='Sales chart' />
                </ListItemButton>
              </ListItem>
              <Divider />
              <ListItem key='test jQuery' disablePadding>
                <ListItemButton onClick={() => navigate('/jQueryTest')}>
                  <ListItemIcon>
                    { '@' }
                  </ListItemIcon>
                  <ListItemText primary='jQuery' />
                </ListItemButton>
              </ListItem>
          </List>
        </Box>
      </SwipeableDrawer>

      <Outlet />
    </>
  );
};

export default App;
