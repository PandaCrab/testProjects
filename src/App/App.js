import React, { useState } from 'react';

import {
  ShippingInfo,
  BillingInfo,
  Payment,
  OrderPlate
} from '../Components/Index';

import {
  Header,
  HeaderText,
  ShoppingBasket,
  Slash,
  Order,
  Route
} from '../Styled/App/AppStyles';
import { GlobalStyles } from '../GlobalStyles';


function App() {
  const [card, setCard] = useState({
    shipping: true,
    billing: false,
    payment: false
  })

const currentForm = () => {
  if (card.shipping === true) return <ShippingInfo />
  else if (card.billing === true) return <BillingInfo />
  else if (card.payment === true) return <Payment />
}

  return (
    <>
      <GlobalStyles />
      <Header>
        <HeaderText> &lt; <Slash>&frasl;</Slash>&gt; Front-end Developer Test Task</HeaderText>
        <ShoppingBasket>cart</ShoppingBasket>

      </Header>
      <Order>
        <Route>
          <a 
          onClick={() => {setCard({
            shipping: true,
            billing: false,
            payment: false}); console.log(card)}}
             value="shipping" href="#Shipping">Shipping</a>&gt;
          <a 
           onClick={() => {setCard({
            shipping: false,
            billing: true,
            payment: false
           }); console.log(card)}}
           value="billing" href="#Billing">Billing</a>&gt;
          <a 
            onClick={ () => {setCard({
              shipping: false,
              billing: false,
              payment: true
            });  console.log(card)} }
            value="payment" href="#Payment">Payment</a>
        </Route>
        {
          currentForm()
        }
        <OrderPlate />
      </Order>

    </>
  );
}

export default App;
