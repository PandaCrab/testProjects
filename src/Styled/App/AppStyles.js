import styled from 'styled-components'
import { devices, colors } from '../../GlobalStyles'

export const Header = styled.section`
  margin: auto;
  margin-bottom: 3em;
  width: 65em;
  height: 5em;
  background-color: ${colors.white};
  display: flex;
  justify-content: space-around;
  border-bottom: 1px solid;
  border-image: ${colors.borderGradient};

  @media ${devices.mobileS} {
    margin-bottom: 1em;
    width: 100%;
    height: 1.3em;

  }

  @media ${devices.tablet} {
    width: 100%;
    height: 2.5em;
  }

  @media ${devices.laptop} {
    width: 65em;
    height: 3em;
  }
`

export const HeaderText = styled.h5`
  margin: auto;
  position: relative;


  @media ${devices.mobileS} {
    font-size: .6em;
  }

  @media ${devices.tablet} {
    font-size: 1em;
  }
`

export const ShoppingBasket = styled.h6`
  margin: auto;
  color: ${colors.slash};

  @media ${devices.mobileS} {
    font-size: .5em;
  }

  @media ${devices.tablet} {
    font-size: .8em;
  }
`

export const Slash = styled.span`
  color: ${colors.slash}; 
  font-weight: 300;
  font-size: 2em;

  @media ${devices.mobileS} {
    font-size: 1.5em;
  }

  @media ${devices.tablet} {
    font-size: 2em;
  }
`

export const Order = styled.div`
  margin: auto;
  width: 45em;
  height: 37em;
  display: flex;
  flex-direction: column;
  background-color: ${colors.white};
  border-radius: 5px;
  box-shadow: 0 0 1.5em #e2e3e5;
  position: relative;

  @media ${devices.mobileS} {
    margin-left: auto;
    width: 275px;
    height: 230px;
    justify-content: flex-start;
  }

  @media ${devices.mobileM} {
    margin-left: auto;
    width: 330px;
    height: 285px;
  }

  @media ${devices.mobileL} {
    margin-left: auto;
    width: 22em;
    height: 17em;
  }

  @media ${devices.tablet} {
    margin-left: auto;
    width: 37em;
    height: 28em;
  }

  @media ${devices.laptop} {
    margin: auto;
    width: 45em;
    height: 37em;
  }
`

export const Route = styled.section`
  max-width: 21em;
  height: 5%;
  margin: 1em;
  display: flex;
  align-items: center;

  @media ${devices.mobileS} {
    font-size: .5em;
  }

  @media ${devices.mobileM} {
    font-size: .5em;
  }

  @media ${devices.mobileL} {
    font-size: .6em;  
  }

  @media ${devices.tablet} {
    font-size: .8em;
  }
`