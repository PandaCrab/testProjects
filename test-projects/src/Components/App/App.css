* {
    margin: 0;
    padding: 0;
}

body {
    background-color: #e6e9f0;
    border-color: linear-gradient(90deg, rgba(235,182,232,1) 0%, rgba(253,185,216,1) 32%, rgba(217,178,248,1) 100%);
}