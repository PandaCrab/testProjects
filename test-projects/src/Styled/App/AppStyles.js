import styled from 'styled-components'

export const Header = styled.section`
  margin: auto;
  margin-bottom: 3em;
  width: 65em;
  height: 3em;
  background-color: #fff;
  display: flex;
  align-items: center;
  justify-content: space-around;
  border-bottom: 1px solid;
  border-image: linear-gradient(to right, #ebb6e8, #fdb9d8, #d9b2f8) 30;
`

export const HeaderText = styled.h5`
  display: flex;
  font-size: 17px;
  color: #8e5bb6;
  align-items: center;
`

export const ShoppingBasket = styled.h6`
  font-size: 15px;
  color: #f03ada;
`

export const Slash = styled.span`
  color: #f03ada; 
  font-weight: 300;
  font-size: 2.3em;
`

export const Order = styled.div`
  margin: auto;
  width: 45em;
  height: 37em;
  display: flex;
  flex-direction: column;
  background-color: #fff;
  justify-content: center;
  border-radius: 5px;
  box-shadow: 0 0 1.5em #e2e3e5;
  position: relative;
`

export const Route = styled.section`
  width: 21em;
  height: 5%;
  margin: 1em;
  display: flex;
  justify-content: space-between;
  align-items: center;
`