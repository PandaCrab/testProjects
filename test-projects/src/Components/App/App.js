import React from 'react';
import styled from 'styled-components';

import ShippingInfo from '../Forms/ShippingInfo'

import './App.css';

const Header = styled.section`
  margin: auto;
  margin-bottom: 3em;
  width: 65em;
  height: 3em;
  background-color: #fff;
  display: flex;
  align-items: center;
  justify-content: space-around;
  border-bottom: 1px solid;
  border-image: linear-gradient(to right, #ebb6e8, #fdb9d8, #d9b2f8) 30;
`

const HeaderText = styled.h5`
  display: flex;
  font-size: 17px;
  color: #8e5bb6;
  align-items: center;
`

const ShoppingBasket = styled.h6`
  font-size: 15px;
  color: #dc63c5;
`

const Slash = styled.span`
  color: #fdb9d8; 
  font-size: 1.7em;
  font-style: italic;
`

const Order = styled.div`
  margin: auto;
  width: 45em;
  height: 37em;
  display: flex;
  flex-direction: column;
  background-color: #fff;
  justify-content: center;
  border-radius: 5px;
  box-shadow: 0 0 1.5em #e2e3e5;
  position: relative;
`

const Info = styled.section`
  margin: 0;
  padding: 2em;
  width: 25em;
  height: 35em;
  position: absolute;
`

const OrderInfo = styled.section`
  margin: 0;
  padding: 2em;
  background-color: #f1f3f6;
  width: 20em;
  height: 37em;
  position: absolute;
  border-radius: inherit;
  top: 0;
  right: 0;
`


function App() {


  return (
    <>
      <Header>
        <HeaderText> &lt;<Slash>&frasl;</Slash> &gt; Front-end Developer Test Task</HeaderText>
        <ShoppingBasket>cart</ShoppingBasket>

      </Header>
      <Order>
        <Info>
            <ShippingInfo />
        </Info>
        <OrderInfo>

        </OrderInfo>
      </Order>

    </>
  );
}

export default App;
