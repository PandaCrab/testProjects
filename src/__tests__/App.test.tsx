import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import App from '../App';

const initialState = {order: {stuff: [
    {
        "id": 1,
        "imgUrl": "https://guesseu.scene7.com/is/image/GuessEU/M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80&op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0",
        "name": "Check Print Shirt",
        "color": "Grey+red+black",
        "price": 110
    }
],
loading: false
}}
const mockStore = configureStore();
let store;

it('should render app component', () => {
    store = mockStore(initialState);
    const component = mount(
            <Provider store={store}>
                <BrowserRouter>
                    <App />
                </BrowserRouter>
            </Provider>
        );
        expect(component).toMatchSnapshot();
});
