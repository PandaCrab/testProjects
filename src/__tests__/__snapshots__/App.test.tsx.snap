// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render app component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      location={
        Object {
          "hash": "",
          "key": "default",
          "pathname": "/",
          "search": "",
          "state": null,
        }
      }
      navigationType="POP"
      navigator={
        Object {
          "action": "POP",
          "back": [Function],
          "block": [Function],
          "createHref": [Function],
          "forward": [Function],
          "go": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <App>
        <Memo(l) />
        <styled.section>
          <section
            className="sc-cCsOjp bhGiXb"
          >
            <styled.h5>
              <h5
                className="sc-jdAMXn eUAjuQ"
              >
                 
                <styled.span>
                  <span
                    className="sc-ciZhAO bjcAgv"
                  >
                    &lt; 
                    <styled.span>
                      <span
                        className="sc-dPyBCJ gYxQKO"
                      >
                        ‚ÅÑ
                      </span>
                    </styled.span>
                    &gt;
                  </span>
                </styled.span>
                 Front-end Developer Test Task
              </h5>
            </styled.h5>
            <styled.section>
              <section
                className="sc-bZnhIo kSpcjZ"
              >
                <styled.p>
                  <p
                    className="sc-cTQhss cXevid"
                  >
                    cart
                  </p>
                </styled.p>
                <styled.div>
                  <div
                    className="sc-iTONeN blezqX"
                  >
                    <Styled(HiOutlineShoppingCart)>
                      <HiOutlineShoppingCart
                        className="sc-jOrMOR bTfVDu"
                      >
                        <IconBase
                          attr={
                            Object {
                              "fill": "none",
                              "stroke": "currentColor",
                              "viewBox": "0 0 24 24",
                            }
                          }
                          className="sc-jOrMOR bTfVDu"
                        >
                          <svg
                            className="sc-jOrMOR bTfVDu"
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
                              key="0"
                              strokeLinecap="round"
                              strokeLinejoin="round"
                              strokeWidth="2"
                            />
                          </svg>
                        </IconBase>
                      </HiOutlineShoppingCart>
                    </Styled(HiOutlineShoppingCart)>
                    <styled.div>
                      <div
                        className="sc-iAvgwm ecDikv"
                      >
                        <styled.p>
                          <p
                            className="sc-efBctP kCdEhA"
                          >
                            1
                          </p>
                        </styled.p>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </section>
            </styled.section>
          </section>
        </styled.section>
        <styled.div>
          <div
            className="sc-bBXxYQ dmLUpr"
          >
            <Outlet />
            <OrderPlate>
              <styled.section
                data-value={false}
                displayMobile="none"
                id="orderInfo"
              >
                <section
                  className="sc-ksZaOG ibhyab"
                  data-value={false}
                  id="orderInfo"
                >
                  <section>
                    <Styled(AiOutlineCloseSquare)
                      id="close-btn"
                      onClick={[Function]}
                    >
                      <AiOutlineCloseSquare
                        className="sc-kLLXSd kBBKOh"
                        id="close-btn"
                        onClick={[Function]}
                      >
                        <IconBase
                          attr={
                            Object {
                              "viewBox": "0 0 1024 1024",
                            }
                          }
                          className="sc-kLLXSd kBBKOh"
                          id="close-btn"
                          onClick={[Function]}
                        >
                          <svg
                            className="sc-kLLXSd kBBKOh"
                            fill="currentColor"
                            height="1em"
                            id="close-btn"
                            onClick={[Function]}
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 1024 1024"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M354 671h58.9c4.7 0 9.2-2.1 12.3-5.7L512 561.8l86.8 103.5c3 3.6 7.5 5.7 12.3 5.7H670c6.8 0 10.5-7.9 6.1-13.1L553.8 512l122.4-145.9c4.4-5.2.7-13.1-6.1-13.1h-58.9c-4.7 0-9.2 2.1-12.3 5.7L512 462.2l-86.8-103.5c-3-3.6-7.5-5.7-12.3-5.7H354c-6.8 0-10.5 7.9-6.1 13.1L470.2 512 347.9 657.9A7.95 7.95 0 0 0 354 671z"
                              key="0"
                            />
                            <path
                              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                              key="1"
                            />
                          </svg>
                        </IconBase>
                      </AiOutlineCloseSquare>
                    </Styled(AiOutlineCloseSquare)>
                    <styled.section>
                      <section
                        className="sc-hAZoDl cQWdqi"
                      >
                        <styled.h6>
                          <h6
                            className="sc-fnykZs kPWXFA"
                          >
                            Order Summary
                          </h6>
                        </styled.h6>
                         
                        <styled.button>
                          <button
                            className="sc-fEOsli ciOOuy"
                          >
                            edit order
                          </button>
                        </styled.button>
                      </section>
                    </styled.section>
                    <styled.div>
                      <div
                        className="sc-bjUoiL gLLEmD"
                      >
                        <Stuff>
                          <StuffItems
                            key="1"
                            stuff={
                              Object {
                                "color": "Grey+red+black",
                                "id": 1,
                                "imgUrl": "https://guesseu.scene7.com/is/image/GuessEU/M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80&op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0",
                                "name": "Check Print Shirt",
                                "price": 110,
                              }
                            }
                          >
                            <styled.section>
                              <section
                                className="sc-hHLeRK dPNuhk"
                              >
                                <styled.img
                                  alt="Check Print Shirt"
                                  src="https://guesseu.scene7.com/is/image/GuessEU/M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80&op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0"
                                >
                                  <img
                                    alt="Check Print Shirt"
                                    className="sc-dmRaPn zCVrD"
                                    src="https://guesseu.scene7.com/is/image/GuessEU/M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80&op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0"
                                  />
                                </styled.img>
                                <styled.section>
                                  <section
                                    className="sc-kgflAQ dbCiuf"
                                  >
                                    <styled.h6>
                                      <h6
                                        className="sc-fLlhyt clxgpt"
                                      >
                                        Check Print Shirt
                                      </h6>
                                    </styled.h6>
                                    <styled.p>
                                      <p
                                        className="sc-bBrHrO bXHGEt"
                                      >
                                        Grey+red+black
                                      </p>
                                    </styled.p>
                                    <styled.p>
                                      <p
                                        className="sc-bBrHrO bXHGEt"
                                      >
                                        Quantity: 1
                                      </p>
                                    </styled.p>
                                  </section>
                                </styled.section>
                                <styled.p>
                                  <p
                                    className="sc-ivTmOn bpCsUh"
                                  >
                                    110
                                    $
                                  </p>
                                </styled.p>
                              </section>
                            </styled.section>
                          </StuffItems>
                        </Stuff>
                      </div>
                    </styled.div>
                    <styled.section>
                      <section
                        className="sc-cxabCf lbMhQR"
                      >
                        <styled.section>
                          <section
                            className="sc-llJcti iyHPPe"
                          >
                            <styled.p>
                              <p
                                className="sc-gicCDI bjluxt"
                              >
                                Subtotal
                              </p>
                            </styled.p>
                            <styled.p>
                              <p
                                className="sc-gicCDI bjluxt"
                              >
                                Shipping
                              </p>
                            </styled.p>
                            <styled.p>
                              <p
                                className="sc-gicCDI bjluxt"
                              >
                                Taxes
                              </p>
                            </styled.p>
                          </section>
                        </styled.section>
                        <styled.section>
                          <section
                            className="sc-iIPllB iIWbyc"
                          >
                            <styled.p>
                              <p
                                className="sc-gicCDI bjluxt"
                              >
                                $
                                110
                              </p>
                            </styled.p>
                            <styled.p>
                              <p
                                className="sc-gicCDI bjluxt"
                              >
                                Free
                              </p>
                            </styled.p>
                            <styled.p>
                              <p
                                className="sc-gicCDI bjluxt"
                              >
                                $
                                12.12
                              </p>
                            </styled.p>
                          </section>
                        </styled.section>
                      </section>
                    </styled.section>
                    <styled.section>
                      <section
                        className="sc-ezWOiH imiMFu"
                      >
                        <styled.p>
                          <p
                            className="sc-bZkfAO gehmlG"
                          >
                            Total
                          </p>
                        </styled.p>
                        <styled.p>
                          <p
                            className="sc-bZkfAO gehmlG"
                          >
                            $
                            122.12
                          </p>
                        </styled.p>
                      </section>
                    </styled.section>
                  </section>
                  <section>
                    <styled.section>
                      <section
                        className="sc-ikZpkk ZNTXr"
                      >
                        <styled.p>
                          <p
                            className="sc-jIZahH jsQqEc"
                          >
                            All purchases are subject to our
                            <Styled(Link)
                              to="/TermsAndConditions"
                            >
                              <Link
                                className="sc-himrzO jeWplj"
                                to="/TermsAndConditions"
                              >
                                <a
                                  className="sc-himrzO jeWplj"
                                  href="/TermsAndConditions"
                                  onClick={[Function]}
                                >
                                  Terms and Conditions
                                </a>
                              </Link>
                            </Styled(Link)>
                          </p>
                        </styled.p>
                      </section>
                    </styled.section>
                  </section>
                </section>
              </styled.section>
              <styled.button
                id="open-close-btn"
                onClick={[Function]}
              >
                <button
                  className="sc-gXmSlM jELbHU"
                  id="open-close-btn"
                  onClick={[Function]}
                >
                  <BsBasket3>
                    <IconBase
                      attr={
                        Object {
                          "fill": "currentColor",
                          "viewBox": "0 0 16 16",
                        }
                      }
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5.757 1.071a.5.5 0 0 1 .172.686L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H.5a.5.5 0 0 1-.5-.5v-1A.5.5 0 0 1 .5 6h1.717L5.07 1.243a.5.5 0 0 1 .686-.172zM3.394 15l-1.48-6h-.97l1.525 6.426a.75.75 0 0 0 .729.574h9.606a.75.75 0 0 0 .73-.574L15.056 9h-.972l-1.479 6h-9.21z"
                          key="0"
                        />
                      </svg>
                    </IconBase>
                  </BsBasket3>
                </button>
              </styled.button>
            </OrderPlate>
          </div>
        </styled.div>
      </App>
    </Router>
  </BrowserRouter>
</Provider>
`;
